pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1150,7 +1150,7 @@ func buildFetchRegistrationEntryQuery(dbType string, req *datastore.FetchRegistr
 func buildFetchRegistrationEntryQuerySQLite3(req *datastore.FetchRegistrationEntryRequest) (string, []interface{}, error) {
 	const query = `
 WITH listing AS (
-	SELECT id FROM registered_entries WHERE entry_id = ?
+	SELECT id FROM registered_entries WHERE entry_id = ? AND (expiry = 0 OR expiry >= strftime('%s','now'))
```

Comment:
a Server User may be checking if some registration(s) exist in the DB with a faster `Read` op to decide if they should Create (totally fresh) or Update (just TTL). These queries now hide this information and forces user to execute `Create` and on `AlreadyExists` execute `Update` (just TTL) instead.